### PR TITLE
Introduce option to enable/disable HOLDLOCK when using MERGE INTO in MS SQL Server

### PR DESCRIPTION
**Pull request by spinatelli**
## Problem

Using "with (HOLDLOCK)" when performing a MERGE INTO statement in MS SQL Server in certain cases brings up performance issues (https://sqlsunday.com/2021/05/04/how-merge-can-deadlock-you/).
The HOLDLOCK is currently hardcoded into the SQL statement of the Sink connector [here](https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java#L327) and cannot be disabled without having a custom JDBC connector.

## Solution
This PR introduces a configuration option (`mssql.use.merge.holdlock`) that allows to enable/disable the use of HOLDLOCK in the MERGE statement. The default is `true`, meaning this PR should be backwards compatible.

NOTE: the unit tests had to be updated, as the JDBCSourceConfig was used everywhere, even when testing Sink features. 

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan
No plan

### CODE REVIEW

File: src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java [hunk 1]
```diff
@@ -324,7 +325,10 @@ public String buildUpsertQueryStatement(
     ExpressionBuilder builder = expressionBuilder();
     builder.append("merge into ");
     builder.append(table);
-    builder.append(" with (HOLDLOCK) AS target using (select ");
+    if (this.config.getBoolean(JdbcSinkConfig.MSSQL_USE_MERGE_HOLDLOCK)) {
```
Comment (aakashnshah) [inline src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java:328]:
```suggestion
    if (this.config.useHoldlockInMerge) {
```

**Review by aakashnshah — COMMENTED** [review]
Looks pretty good overall, I'm not sure if we want to a add configuration specifically for one dialect and if there are any alternatives.

File: src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java (hunk 1 above)
Comment (spinatelli) [reply src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java]: Needs casting to JdbcSinkConfig

((JdbcSinkConfig) this.config).useHoldlockInMerge

**Comment by spinatelli** [comment]
Hi Aakash, thanks for the suggestion and the code check.
As for the dialect-specific configuration, if there are any alternatives that are better suited I am happy to update the code with the needed implementation.
Would be great to have an indication about that :)

**Comment by spinatelli** [comment]
Hi @aakashnshah, do you have any further input from your side about the dialect-specific configurations? :)

**Review by ManasjyotiSharma — APPROVED** [review]
LGTM; awesome work! Thanks @spinatelli for your efforts here.

**Pull request merged**